src/tests/ftest/server/dynamic_start_stop.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+import json
+from apricot import TestWithServers
+from dmg_utils import DmgCommand
+
+
+class DynamicStartStop(TestWithServers):
+    """Test Class Description:
+
+    SRS-10-0052 is about stopping the server and SRS-10-0053 is about starting,
+    so this test focuses on stopping and starting the new servers in an existing
+    system with a pool.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DynamicStartStop object."""
+        super(DynamicStartStop, self).__init__(*args, **kwargs)
+        self.dmg_cmd = None
+        self.stopped_ranks = set()
+
+    def verify_system_query(self):
+        """Verify state of the ranks.
+
+        Call dmg system query --json and verify the State of each rank. If the
+        rank is in self.stopped_ranks, verify that its status is Stopped.
+        Otherwise, Joined.
+        """
+        output = self.dmg_cmd.system_query().stdout
+        data = json.loads(output)
```

Comment:
I'm wondering if we don't want to update the `DmgCommand.system_quey()` method return the dictionary (`data`), where w/o JSON output it could manually populate the dictionary with regex results, but with `self.json.value` set it would assign `data = json.loads(output)`.  This would enable the test to run with either JSON or normal output from the `dmg system query` command w/o modification - assuming the manual `data` population uses the same keys.  We could also document what these keys and dictionary structure in the `DmgCommand.system_quey()` method.

Reply:
I understand and I think supporting both json and string output is the ideal implementation. However, I'm not sure if building the regex and the data structure are high priority tasks because we don't need to worry about them if we use json.

Reply:
I understand the priority argument.  One concern, and maybe not directly related to this PR, is are we verifying that the stdout and json output provide the same information somewhere?